### PR TITLE
@igosoft patches merge — second batch 

### PR DESCRIPTION
**Pull request by ChALkeR**
Ref: #128, #62.

This one is small, hoping for a quick review here.

/cc @igosoft @Plaristote @akreuzkamp

### CODE REVIEW

File: src/qtcore/qml/UpdateGeometry.js [hunk 1]
```diff
@@ -194,6 +196,38 @@ function updateVGeometry(newVal, oldVal, propName) {
...
+    var minY = children.length>0 ? children[0].y : 0;
+    var h=0;
+    var w=0;
+    var child;
+
+    for (var i=0;i<children.length; i++){
+        child = children[i];
+
+        h = child.$isUsingImplicitHeight ? child.implicitHeight : child.height;
+        w = child.$isUsingImplicitWidth ? child.implicitWidth : child.width;
```
Comment (akreuzkamp) [inline src/qtcore/qml/UpdateGeometry.js]: This shouldn't be necessary, unless you check values manually using child.$properties.implicitHeight.value.
child.height already gives you it's implicitHeight, if it's using that.

Comment (ChALkeR) [inline src/qtcore/qml/UpdateGeometry.js]: @akreuzkamp, it is also used in a many more places at c1bf9083d7beabf53d676ec2f187f82cde973e97.

Are all those `child.$isUsingImplicitHeight ? child.implicitHeight : child.height` and `child.$isUsingImplicitWidth ? child.implicitWidth : child.width` unnecessary?

/cc @igosoft

File: src/qtcore/qml/elements/QtQuick.Controls/ScrollView.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  name:     'ScrollView',
+  versions: /.*/,
+  baseClass: QMLItem,
+  constructor: function QMLScrollView(meta) {
+    QMLItem.call(this, meta);
+
+    var self = this;
+    
+ 
+    this.dom.style.pointerEvents = "auto";
```
Comment (akreuzkamp) [inline src/qtcore/qml/elements/QtQuick.Controls/ScrollView.js]: you can use this.css ;)

File: src/qtcore/qml/UpdateGeometry.js (hunk 1 above)
Comment (igosoft) [inline src/qtcore/qml/UpdateGeometry.js]: @akreuzkamp @ChALkeR I can get rid of those "checks" if they aren't necessary. Please let me know.

Comment (akreuzkamp) [inline src/qtcore/qml/UpdateGeometry.js]: If $isUsingImplicitHeight is set, implicitHeight and height are synchronized. That's the idea of implicitHeight, after all ;)
So yes, I'm pretty sure that all of these checks are unnecessary (except of setting css.height to auto, but that's something different anyway :) )

File: src/qtcore/qml/elements/QtQuick/MouseArea.js [hunk 3]
```diff
@@ -81,6 +83,16 @@ registerQmlType({
...
+    this.dom.onmouseover = function(e) {
+        self.containsMouse = true;
+        self.dom.style.cursor = cursorShapeToCSS();
+        self.entered();
+    }
+    this.dom.onmouseout = function(e) {
+        self.containsMouse = false;
+        self.dom.style.cursor = "auto";
+        self.exited();
+    }
```
Comment (Plaristote) [inline src/qtcore/qml/elements/QtQuick/MouseArea.js]: I don't understand why we have to do this. If you simply watch for the `cursorShape` property, and set the `dom.style.cursor` value whenever it changes, won't it do the same thing ? Do we really need to update the css attribute everytime the mouse enters or exits it ?

Comment (igosoft) [inline src/qtcore/qml/elements/QtQuick/MouseArea.js]: Indeed, changing style here doesn't make sense :(. I can't remember why I did such a change :). I think we can remove these lines (self.dom.style.cursor). Do you want me to remove these lines now ?

Comment (Plaristote) [inline src/qtcore/qml/elements/QtQuick/MouseArea.js]: I've made a commit to update the behavior. I also found an issue with the `pressedButtons` property: in an event handler, it was refered as `this.pressedButtons`. But inside the event handlers, `this` isn't the QML component: that's why we have the `self` variable around, sometimes. I've also made a commit to fix that.

Comment (ChALkeR) [inline src/qtcore/qml/elements/QtQuick/MouseArea.js]: Yes, I missed that to. That should have been an arrow function, transplited by babel =).
But lets use `self` for now.

File: src/qtcore/qml/elements/QtQuick/MouseArea.js [hunk 4]
```diff
@@ -89,5 +91,41 @@ registerQmlType({
...
+          case Qt.ClosedHandCursor: cursor = "move";break;
+          case Qt.DragCopyCursor: cursor = "auto";break;
+          case Qt.DragMoveCursor: cursor = "auto";break;
+          case Qt.DragLinkCursor: cursor = "auto";break;
+          case Qt.LastCursor: cursor = "auto";break;
+          case Qt.BitmapCursor: cursor = "auto";break;
+          case Qt.CustomCursor: cursor = "auto";  break;
+        }
+        return cursor;
+    }
```
Comment (ChALkeR) [inline src/qtcore/qml/elements/QtQuick/MouseArea.js]: btw, this could be simplified a lot. Will add an additional commit.

**Comment by ChALkeR** [comment]
@igosoft I'm going to squash your updates here.

**Comment by ChALkeR** [comment]
I added 7fba33b6ea04dd7879eddaae7bf8b021699b13ad, it implements all those cursors, excluding the custom ones and the `LastCursor` (which is now an alias to DragLinkCursor).

Not sure what `LastCursor` does and where it's documented, btw.

File: src/qtcore/qml/UpdateGeometry.js [hunk 5]
```diff
@@ -95,6 +95,8 @@ function updateHGeometry(newVal, oldVal, propName) {
         this.width = width;
 
     this.$updatingGeometry = false;
+
+    if (this.parent != undefined) updateChildrenRect(this.parent);
```
Comment (ChALkeR) [inline src/qtcore/qml/UpdateGeometry.js]: This is the same as `if (this.parent)`. It should be either that or `if (this.parent !== undefined)`.

File: src/qtcore/qml/UpdateGeometry.js [hunk 6]
```diff
@@ -194,6 +196,32 @@ function updateVGeometry(newVal, oldVal, propName) {
         this.height = height;
 
     this.$updatingGeometry = false;
+
+    if (this.parent != undefined) updateChildrenRect(this.parent);
```
Comment (ChALkeR) [inline src/qtcore/qml/UpdateGeometry.js]: Ditto.

File: src/qtcore/qml/UpdateGeometry.js (hunk 5 above)
Comment (igosoft) [inline src/qtcore/qml/UpdateGeometry.js]: @ChALkeR done

**Comment by ChALkeR** [comment]
LGTM

Will squash two commits a bit later, though.

**Comment by ChALkeR** [comment]
Squashed, added one more commit on top of that to de-duplicate some code in ScrollView.
Everything here LGTM, waiting for more reviews.

**Comment by Plaristote** [comment]
LGTM

**Pull request merged**